testbed/tests/window/test_window.py

```diff
@@ -142,13 +143,123 @@ async def test_move_and_resize(main_window, main_window_probe, capsys):
-        """Window can be made full screen"""
-        main_window.full_screen = True
-        await main_window_probe.wait_for_window("Full screen is a no-op")
+    async def test_window_state_minimized(main_window, main_window_probe):
+        """Window can have minimized window state"""
+        assert main_window_probe.get_window_state() == WindowState.NORMAL
+        assert main_window_probe.get_window_state() != WindowState.MINIMIZED
+        main_window.state = WindowState.MINIMIZED
+        await main_window_probe.wait_for_window("WindowState.MINIMIZED is a no-op")
+        assert main_window_probe.get_window_state() != WindowState.MINIMIZED
+        assert main_window_probe.get_window_state() == WindowState.NORMAL
+
+    async def test_window_state_maximized(main_window, main_window_probe):
+        """Window can have maximized window state"""
+        assert main_window_probe.get_window_state() == WindowState.NORMAL
+        assert main_window_probe.get_window_state() != WindowState.MAXIMIZED
+        main_window.state = WindowState.MAXIMIZED
+        await main_window_probe.wait_for_window("WindowState.MAXIMIZED is a no-op")
+        assert main_window_probe.get_window_state() != WindowState.MAXIMIZED
+        assert main_window_probe.get_window_state() == WindowState.NORMAL
+
+    async def test_window_state_full_screen(main_window, main_window_probe):
+        """Window can have full screen window state"""
+        assert main_window_probe.get_window_state() == WindowState.NORMAL
+        assert main_window_probe.get_window_state() != WindowState.FULLSCREEN
+        # Make main window full screen
+        main_window.state = WindowState.FULLSCREEN
+        await main_window_probe.wait_for_window("Main window is full screen")
+        assert main_window_probe.get_window_state() == WindowState.FULLSCREEN
+        assert main_window_probe.get_window_state() != WindowState.NORMAL
+
+        main_window.state = WindowState.FULLSCREEN
+        await main_window_probe.wait_for_window("Main window is still full screen")
+        assert main_window_probe.get_window_state() == WindowState.FULLSCREEN
+        assert main_window_probe.get_window_state() != WindowState.NORMAL
+        # Exit full screen
+        main_window.state = WindowState.NORMAL
+        await main_window_probe.wait_for_window("Main window is not full screen")
+        assert main_window_probe.get_window_state() != WindowState.FULLSCREEN
+        assert main_window_probe.get_window_state() == WindowState.NORMAL
+
+        main_window.state = WindowState.NORMAL
+        await main_window_probe.wait_for_window("Main window is still not full screen")
+        assert main_window_probe.get_window_state() != WindowState.FULLSCREEN
+        assert main_window_probe.get_window_state() == WindowState.NORMAL
+
+    async def test_window_state_presentation(main_window, main_window_probe):
+        # WindowState.PRESENTATION is implemented on Android but not on iOS.
```

Comment:
In which case, the test should be skipped, not worked around.

Reply:
I have added a skipped decorator for iOS. Also, if a feature doesn't exist at all on mobile platforms, like `WindowState.MINIMIZED`- doesn't exist both for Android and iOS, then should there be a test checking that setting `WindowState.MINIMIZED` is a no-op or there should not be any test for it at all?